services/connector/context_test.go

```diff
@@ -0,0 +1,59 @@
+package connector
+
+import (
+	"context"
+	"testing"
+)
+
+func TestConnectionType(t *testing.T) {
+	tests := []struct {
+		name          string
+		connType      ConnectionType
+		wantUntrusted bool
+		wantTrusted   bool
+	}{
+		{
+			name:          "HTTP connection is untrusted",
+			connType:      ConnectionTypeHTTP,
+			wantUntrusted: true,
+			wantTrusted:   false,
+		},
+		{
+			name:          "Internal connection is trusted",
+			connType:      ConnectionTypeInternal,
+			wantUntrusted: false,
+			wantTrusted:   true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := WithConnectionType(context.Background(), tt.connType)
+
+			if got := IsUntrustedConnection(ctx); got != tt.wantUntrusted {
+				t.Errorf("IsUntrustedConnection() = %v, want %v", got, tt.wantUntrusted)
+			}
+
+			if got := IsTrustedConnection(ctx); got != tt.wantTrusted {
+				t.Errorf("IsTrustedConnection() = %v, want %v", got, tt.wantTrusted)
+			}
+
+			if got := GetConnectionType(ctx); got != tt.connType {
+				t.Errorf("GetConnectionType() = %v, want %v", got, tt.connType)
+			}
+		})
+	}
```

Comment:
Mate, keep it simple 😄 
```suggestion
    ctx := WithConnectionType(context.Background(), ConnectionTypeHTTP)
    require.False(IsTrustedConnection(ctx))
    
    ctx = WithConnectionType(context.Background(), ConnectionTypeInternal)
    require.True(IsTrustedConnection(ctx))
```

Reply:
Indeed:) 

Added more tests for trusted connection + empty clientID check